src/noLifeCycleCallRule.ts

```diff
@@ -14,7 +14,8 @@ export class Rule extends Lint.Rules.AbstractRule {
     typescriptOnly: true,
   };
 
-  static FAILURE_STRING: string = 'Avoid explicitly calls to lifecycle hooks in class "%s"';
+  static FAILURE_STRING_FOR_CLASS: string = 'Avoid explicit calls to lifecycle hooks in class "%s".';
+  static FAILURE_STRING: string = 'Avoid explicit calls to lifecycle hooks.';
```

Comment:
I think you can only have one `FAILURE_STRING` (and let's remove the unnecessary `:string`):

`static FAILURE_STRING = 'Avoid explicit calls to lifecycle hooks.';`

Reply:
You can have more than one.

Reply:
I didn't mention that it's impossible. I'm suggesting that you remove and keep only one failure. :)

Reply:
I can remove the type annotation. I kept it because it was already there.

Reply:
But I wanted to keep the class name when it's available.

Reply:
I have updated this.